docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+1. The command execution can affect the `Model` (e.g. setting a value).
+1. In addition, the `Ui` may also perform certain actions, such as displaying help to the user.
+
+### Model component
+
+![ModelDiagram](diagrams/ModelClassDiagram.png)
+
+<small><i>Figure 5</i></small>
+
+The `Model`,
+* includes `CircuitTemplate` and `BooleanTemplate` that can represent the current `template` in [`Logic`](#logic-component).
+* has `Component` and `Gate` within the templates.
+* does not depend on any of the other three components.
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Implementation of Circuit Commands (Dash) <a name="circ-comd"></a>
+
+This section provides details on the implementation of the various electronic circuit commands.
+
+There are 4 different types of components of electronic circuits that can be instantiated in the program:
+* `Resistor` - A _resistor_ component.
+* `Capacitor` - A _capacitor_ component.
+* `Inductor` - An _inductor_ component.
+* `VoltageSource` - An _Alternating Current_ Voltage Source.
+
+Each component is used within a circuit template, based on the kind of circuit instantiated.
+
+<div markdown="span" class="alert alert-primary">
+
+:information_source: **Note:** The `VoltageSource` is instantiated in all Circuit Templates.
+
+</div>
+
+There are four different circuit templates that can be instantiated in the program:
+
+* `LcTemplate` - Inductor-Capacitor Circuit Template
+* `RTemplate` - Resistor Circuit Template
+* `RcTemplate` - Resistor-Capacitor Template (extends `RTemplate`)
+* `LrTemplate` - Inductor-Resistor Template (extends `RTemplate`)
+
+![CircuitCommandClass](diagrams/CircuitCommandClassDiagram.png)
+
+<small><i>Figure 6</i></small>
+
+The diagram above demonstrates the relationship between the various `CircuitCommand` objects. The various commands to be parsed are as explained in this section. While the **User Guide** explains the commands used on the CLI, this section goes into detail the classes used to execute the commands. The command classes also make use of the classes in the [**`Model`**](#model-component) shown in the diagram below when executing the commands.
+
+![CircuitModelClass](diagrams/CircuitModelClassDiagram.png)
+
+<small><i>Figure 7</i></small>
+
+### `TemplateCircuitCommand`  <a name="temp-circ"></a>
+Given below is the sequence diagram for interactions within the `logic` and `template` components for the 
+`parse(template rl)` API call that implements the `template` command to create templates.
+
+![TemplateSequence](diagrams/TemplateSequenceDiagram.png)
+
+<small><i>Figure 8</i></small>
+
+### `SetCircuitCommand` <a name="set-circ"></a>
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parse("set r 500")` API call which implements the `set` command to set values of components.
+
+![SetSequence](diagrams/SetSequenceDiagram.png)
```

Comment:
![Screen Shot 2020-10-29 at 2 38 34 PM](https://user-images.githubusercontent.com/47135468/97534228-96580e80-19f4-11eb-9dc8-0cdd0178f994.png)

Really like this diagram, and how the different components are clearly demarcated here. Here, does `Parser` communicate with `Template` before the `SetCommand(template, "r", ...)` call? Or is this abstracted away in this diagram? Also, would some textual description relating the previous `TemplateCircuitCommand` sequence diagram to this diagram be good, to elaborate the relation/dependency between creating a template and setting components of templates?

Reply:
Abstracted away in https://github.com/AY2021S1-CS2113T-W13-3/tp/pull/121, thanks!

![GetCommand](https://user-images.githubusercontent.com/53231968/97583864-5fa0e900-1a32-11eb-80f8-1698477d6239.png)